server/src/test/java/io/crate/expression/scalar/ArrayAppendFunctionTest.java

```diff
@@ -0,0 +1,205 @@
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar;
+
+import io.crate.expression.symbol.Literal;
+import io.crate.types.ArrayType;
+import io.crate.types.DataTypes;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.crate.testing.SymbolMatchers.isFunction;
+import static io.crate.testing.SymbolMatchers.isLiteral;
+
+public class ArrayAppendFunctionTest extends ScalarTestCase {
+
+    private static final ArrayType<Integer> INTEGER_ARRAY_TYPE = new ArrayType<>(DataTypes.INTEGER);
+
+    @Test
+    public void testNormalizeWithValueSymbols() throws Exception {
+        assertNormalize("array_append([10, 20], 30)", isLiteral(Arrays.asList(10, 20, 30)));
+    }
+
+    @Test
+    public void testNormalizeWithTextValueSymbols() throws Exception {
+        assertNormalize("array_append(['10', '20'], '30')", isLiteral(Arrays.asList("10", "20", "30")));
+    }
+
+    @Test
+    public void testNormalizeWithRefs() throws Exception {
+        assertNormalize("array_append(long_array, 30)", isFunction(ArrayAppendFunction.NAME));
+    }
+
+    @Test
+    public void testNullObjectArgument() throws Exception {
+        assertNormalize("array_append([1, 2, 3], null)", isLiteral(Arrays.asList(1, 2, 3, null)));
+    }
+
+    @Test
+    public void testNullArrayArgument() throws Exception {
+        assertNormalize("array_append(null, 1)", isLiteral(List.of(1)));
+    }
+
+    @Test
+    public void testNullArrayInnerTypeArgument() throws Exception {
+        assertNormalize("array_append([null], 1)", isLiteral(Arrays.asList(null, 1)));
+    }
+
+    @Test
+    public void testOneNullArrayInnerTypeArgumentNullObject() throws Exception {
+        assertNormalize("array_append([null, 1], null)", isLiteral(Arrays.asList(null, 1, null)));
+    }
+
+    @Test
+    public void testEmptyArrayArgument() throws Exception {
+        assertNormalize("array_append([], 1)", isLiteral(List.of(1)));
+    }
+
+    @Test
+    public void testDifferentConvertableInnerTypesLong() throws Exception {
+        assertNormalize("array_append([1::integer], 2::long)", isLiteral(Arrays.asList(1L, 2L)));
+    }
+
+    @Test
+    public void testCastTextToIntegerInnerTypes() throws Exception {
+        assertNormalize("array_append([1,2], '3')", isLiteral(Arrays.asList(1, 2, 3)));
+    }
+
+    @Test
+    public void testCastTextToDoubleInnerTypes() throws Exception {
+        assertNormalize("array_append([1,2.0], '3.0')", isLiteral(Arrays.asList(1.0, 2.0, 3.0)));
+    }
+
+    @Test
+    public void testDifferentConvertableInnerTypesDouble() throws Exception {
+        assertNormalize("array_append([1::integer], 2::double)", isLiteral(Arrays.asList(1.0, 2.0)));
+    }
```

Comment:
I think one test to ensure that implicit casting of inner types is allowed for this scalar is enough.

Reply:
Done